ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/map.openlayers.js

```diff
@@ -744,7 +778,34 @@ module.exports = function OpenlayersMap(
          Updates a passed in geometry to be shown
          */
     showGeometry(geometry) {
-      geometry.setVisible(true)
+      const feature = geometry.getSource().getFeatures()[0]
+      if (feature.getProperties().isLabel) {
+        // only show one label at one location
+        const geometryInstance = feature.getGeometry()
+        const labelWithSamePosition = _.find(
+          mapModel.get('labels'),
+          label =>
+            label
+              .getSource()
+              .getFeatures()[0]
+              .getGeometry()
+              .getCoordinates()[0] === geometryInstance.getCoordinates()[0] &&
+            label
+              .getSource()
+              .getFeatures()[0]
+              .getGeometry()
+              .getCoordinates()[1] === geometryInstance.getCoordinates()[1] &&
+            (label.get('isSelected') || label.getVisible())
+        )
+        if (
+          labelWithSamePosition === undefined ||
+          geometry.get('id') === labelWithSamePosition.get('id')
+        ) {
+          geometry.setVisible(true)
+        }
+      } else {
+        geometry.setVisible(true)
+      }
```

Comment:
❓ Can you make this a helper method that takes in a `feature`? It looks like it's used multiple times above.

Reply:
The only reason I haven't is because the logic differs slightly every time. I can see if I can factor anything out though.